saturn-gpu/environment.yml

```diff
@@ -40,11 +40,11 @@ dependencies:
 - pyarrow
 - python=3.7
 - pytorch=1.7.0=py3.7_cuda10.1.243_cudnn7.6.3_0
-- rapids
+- rapids=0.17.0=cuda10.1_py37_g180f433_165
 - s3fs
 - scikit-learn
 - scipy
-- snowflake-connector-python==2.3.2
+- snowflake-connector-python==2.1.1
```

Comment:
This seemed to be the only way to resolve this env. I don't think it'll pose an issue, but @jameslamb probably knows better.

Reply:
2.1.1 is very old (December 2019): https://github.com/snowflakedb/snowflake-connector-python/releases/tag/v2.1.1, which makes me nervous that it might not actually work with current versions of Snowflake. I don't know how Snwoflake manages versioning their API and client together.

I also think this change would make building a custom image based on `saturn/saturn-gpu` a really bad one. Look at all of these version ceilings https://github.com/snowflakedb/snowflake-connector-python/blob/7f938d2a10b0a055c62e07f410266e3cba93df27/setup.py#L164. :tired_face: 

I think `snowflake-connector-python` is a so difficult to build that we should either:

* just support a different image like `saturncloud/saturn-snowflake-gpu`
* go try to contribute to https://github.com/snowflakedb/snowflake-connector-python to help it relax its dependency constraints

Reply:
I definitely agree that we should separate out snowflake into its own image. That'll be part of the big image split. I guess to test this out I need to build the image and try to run the example notebooks.

Reply:
I'm personally against pinning to a snowflake version from 2019, even if you build an image and the examples work, because of all those version ceilings they put on popular packages. Those ceilings will make building custom images off of this one tough.

Reply:
The reason why we are pinning to such an old version is because the ceilings in that package aren't as strict. So yes it's not great, but I think it actually makes it easier for the resolve.

Reply:
I moved it to the pip section since the pip package doesn't have the tight pins on pandas/pyarrow. I think that's the best we can do for an env that has both snowflake-connector-python and rapids (which pins pandas as well).

Reply:
ok, sounds good